pkg/kapp/crdupgradesafety/validator.go

```diff
@@ -91,3 +93,30 @@ func NoStoredVersionRemoved(old, new v1.CustomResourceDefinition) error {
 
 	return nil
 }
+
+func NoExistingFieldRemoved(old, new v1.CustomResourceDefinition) error {
+	reg := manifestcomparators.NewRegistry()
+	err := reg.AddComparator(manifestcomparators.NoFieldRemoval())
+	if err != nil {
+		return err
+	}
+
+	results, errs := reg.Compare(&old, &new)
+	if len(errs) > 0 {
+		return errors.Join(errs...)
+	}
+
+	errSet := []error{}
+
+	for _, result := range results {
+		if len(result.Errors) > 0 {
+			errSet = append(errSet, errors.New(strings.Join(result.Errors, "\n")))
+		}
+	}
+	if len(errSet) > 0 {
+		baseErr := errors.New("existing field in the CRD schema cannot be removed for safe CRD upgrades")
```

Comment:
Nit: I'm not sure this base error is needed. The top level validator this is intended to be used with should return which individual validator failed, and looking at the errors returned by the crd-schema-checker package for the NoFieldRemoval comparator it returns an error of the format `crd/%v version/%v field/%v may not be removed` which seems like it would be descriptive enough.

WDYT?

Reply:
Addressed in [latest commit](https://github.com/carvel-dev/kapp/pull/922/commits/51b64005ce47e75e822f08f1bda3e656320bf0c3) as well. Originally added it to be able to use it to assert in the e2e test but ended up using the error returned by crd-schema-checker package, so the `baseErr` is not required anymore.